Doppler.BigQueryMicroservice/Controllers/BigQueryController.cs

```diff
@@ -41,14 +58,38 @@ public async Task<ActionResult> GetAllowedEmails(string accountName)
         [HttpPut("/big-query/{accountName}/allowed-emails")]
         public async Task<ActionResult> SaveAllowedEmails(string accountName, AllowedEmails model)
         {
-            var user = await _userRepository.GetUserByEmail(accountName);
-            if (user == null)
+            try
             {
-                return NotFound();
+                bool notified = false;
+                var user = await _userRepository.GetUserByEmail(accountName);
+                if (user == null)
+                {
+                    return NotFound();
+                }
+
+                var results = await _userAccessByUserRepository.MergeEmailsAsync(user.IdUser, model.Emails);
+                var newEmails = results.Where(a => a.Action == "INSERT").Select(e => e.Email).ToList();
+                var template = _emailSettings.Value.BigQueryInvitationTemplateId[model.Languaje.ToLower()];
+                if (newEmails.Count > 0)
+                {
+                    try
+                    {
+                        notified = await _emailSender.SendWithTemplateAsync(template, newEmails, model.Emails);
+                    }
+                    catch (System.Exception ex)
+                    {
+                        _logger.LogError(ex, "Error in method SendWithTemplateAsync, service {service}", nameof(EmailSender));
+                        notified = false;
+                    }
```

Comment:
Capturing exceptions in some notifications is a common scenario, what do you think about moving try/catch logic to the email sender?

```suggestion
                    notified = await _emailSender.SafeSendWithTemplateAsync(template, newEmails, model.Emails);
```

In other Doppler's projects we are using the prefix `Safe` to indicate that a method does not throw exceptions.